server/src/main/java/org/elasticsearch/transport/TransportRequestOptions.java

```diff
@@ -19,18 +19,32 @@
 
 package org.elasticsearch.transport;
 
+import org.elasticsearch.common.Nullable;
 import org.elasticsearch.common.unit.TimeValue;
 
 public class TransportRequestOptions {
 
+    @Nullable
     private final TimeValue timeout;
     private final Type type;
 
-    private TransportRequestOptions(TimeValue timeout, Type type) {
+    public static TransportRequestOptions of(@Nullable TimeValue timeout) {
```

Comment:
I'd prefer this wasn't called `of` since this suggests there aren't any other options (and then it'd be weird to call it "options"). Does `timeout()` work?

Marks for brevity tho :) Not a strongly-held opinion, I can live with it.